batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/service/StudentService.java

```diff
@@ -20,16 +23,30 @@
     return studentDao.fetch(offset, limit);
   }
 
-  public void insert(Student student) throws DataException {
-    studentDao.insert(student);
+  public void insert(Student student) throws DataException, ValidationException {
+    StudentValidator validator = new StudentValidator();
+    Map<String, String> errors = validator.validate(student);
+
+    if(errors.isEmpty()){
+      studentDao.insert(student);
+    }else {
+      throw new ValidationException("validation failed", errors);
+    }
   }
 
   public void delete(Integer studentId) throws DataException {
     studentDao.delete(studentId);
   }
 
-  public void update(Student student) throws DataException {
-    studentDao.update(student);
+  public void update(Student student) throws DataException, ValidationException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: !unknownSymbol!, !unknownSymbol! [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1160)